src/preamp.jl

```diff
@@ -25,18 +25,40 @@ This struct is currently mutable because in the case of noise modelling based on
     τ_rise::typeof(1.0*ns_unit) = T(15)*u"ns"
 
     "Preamp offset in keV"
-    offset::typeof(1.0*energy_unit) = 0u"keV";
+    offset_keV::typeof(1.0*energy_unit) = 0u"keV";
 
-    "Maximum DAQ energy"
-    max_e::typeof(1.0*energy_unit) = 10_000u"keV" # == typemax(UInt16)
+    "Preamp offset in ADC (alternative)"
+    offset_ADC = 0; # ADC units
+
+    # "Maximum DAQ energy"
+    # max_e::typeof(1.0*energy_unit) = 0u"keV" # == typemax(UInt16)
 
     "Preamp gain"
     # If offset = 0, it means gain has to be inferred (from the data baseline offset)
     # How does it end up being Float64?
-    gain = offset == 0u"keV" ? 0 : typemax(UInt16) / ((max_e+offset)/uconvert(u"keV", germanium_ionization_energy))
+    # gain =             
+    #     if offset_keV == 0u"keV"
+    #         if offset_ADC == 0
+    #             0
+    #         else
+    #             # ( typemax(UInt16) - offset_ADC ) * uconvert(u"keV", germanium_ionization_energy) / max_e # ADC/charge
+    #             ( typemax(UInt16) - offset_ADC ) / uconvert(u"eV", max_e) # ADC/eV
+    #         end
+    #     else
+    #         # typemax(UInt16) / ((max_e+offset_keV)/uconvert(u"keV", germanium_ionization_energy)) # ADC/charge
+    #         typemax(UInt16) / uconvert(u"eV", max_e + offset_keV) # ADC/eV
+    #     end            
+    gain::typeof(1.0/1u"eV")
+    # gain = offset == 0u"keV" ? 0 : typemax(UInt16) / ((max_e+offset)/uconvert(u"keV", germanium_ionization_energy))
+    # gain = offset_ADC == 0 ? : ( typemax(UInt16) - offset_ADC ) * uconvert(u"keV", germanium_ionization_energy) / max_e
+
+    "Preamp Gaussian noise in ADC"
+    noise_σ_ADC = 0
+
+    "Preamp Gaussian noise in keV"
+    noise_σ_keV::typeof(1.0*energy_unit)= 0u"keV"
```

Comment:
It might be helpful to note somewhere that if both values are given, `noise_σ_keV` has priority and will be the value used.

Reply:
Yes I made a ToDo comment for myself somewhere to add a check, if both are added to either throw an error or a warning. Same for offset in keV or ADC, big warning if both are provided, both will be double added.